ambry-router/src/test/java/com.github.ambry.router/GetBlobOperationTest.java

```diff
@@ -530,16 +596,97 @@ public void run() {
+   * @param callback the callback to run after completion of the operation, or {@code null} if no callback.
+   * @return the operation
+   * @throws Exception
+   */
+  private GetBlobOperation createOperationAndComplete(Callback<ReadableStreamChannel> callback)
+      throws Exception {
+    return createOperationAndComplete(callback, null);
+  }
+
+  /**
+   * Create a getBlob operation with the specified callback and poll until completion.
+   * @param callback the callback to run after completion of the operation, or {@code null} if no callback.
+   * @param opReference if non-null, this will point to the {@link GetBlobOperation} right after creation
+   * @return the operation
+   * @throws Exception
+   */
+  private GetBlobOperation createOperationAndComplete(Callback<ReadableStreamChannel> callback,
+      AtomicReference<GetBlobOperation> opReference)
+      throws Exception {
+    GetBlobOperation op = createOperation(callback, opReference);
+    while (!op.isOperationComplete()) {
+      op.poll(requestRegistrationCallback);
+      List<ResponseInfo> responses = sendAndWaitForResponses(requestRegistrationCallback.requestListToFill);
+      for (ResponseInfo response : responses) {
+        op.handleResponse(response);
+      }
+    }
+    return op;
+  }
+
+  /**
+   * Create a getBlob operation with the specified callback
+   * @param callback the callback to run after completion of the operation, or {@code null} if no callback.
+   * @return the operation
+   * @throws Exception
+   */
+  private GetBlobOperation createOperation(Callback<ReadableStreamChannel> callback)
+      throws Exception {
+    return createOperation(callback, null);
+  }
+
+  /**
+   * Create a getBlob operation with the specified callback
+   * @param callback the callback to run after completion of the operation, or {@code null} if no callback.
+   * @param opReference if non-null, this will point to the {@link GetBlobOperation} right after creation
+   * @return the operation
+   * @throws Exception
+   */
+  private GetBlobOperation createOperation(Callback<ReadableStreamChannel> callback,
+      AtomicReference<GetBlobOperation> opReference)
+      throws Exception {
+    operationsCount.incrementAndGet();
+    GetBlobOperation op =
+        new GetBlobOperation(routerConfig, routerMetrics, mockClusterMap, responseHandler, blobIdStr, operationFuture,
+            callback, operationCompleteCallback, readyForPollCallback, blobIdFactory, time);
+    requestRegistrationCallback.requestListToFill = new ArrayList<>();
+    if (opReference != null) {
+      opReference.set(op);
+    }
+    return op;
+  }
+
+  /**
+   * Check that an operation is complete and has the specified {@link RouterErrorCode} set.
+   * @param op The operation to check.
+   * @param expectedError The error code expected.
+   */
+  private void checkErrorCode(GetBlobOperation op, RouterErrorCode expectedError) {
```

Comment:
Since these methods are asserting failure and the error code will not be null, do you think you should name this `assertFailureAndCheckErrorCode()`? Here and for similar methods in other tests?